Allow to set the PUID and PGID using docker

Macavity opened the pull request:
## Feature 

This solves the request #9707 

### **Description of the Issue:**

The current Dockerfile hardcodes the creation of a user and group with specific user IDs (UID) and group IDs (GID). This approach lead to two major problems:

1. **Unknown groups in the container:** When mounting directories from the host, if the GID of the mounted directory belongs to a group unknown to the container, permission issues arise, preventing the container from accessing the directory.
2. The creation of the "/.config/" directory required root permissions, which the siyuan user does not have.


### **Changes Made:**

- Provide ARGs for docker-compose to allow overwriting the default 1000:1000 values
- Dynamically assigns the `siyuan` user in case the provided GID exists already on the container
- Ensured that mounted directories, such as the workspace, are accessible by the container user by aligning the user's group membership with host permissions.
- Adjusted ownership of the relevant directories to ensure correct group-based permissions.

Macavity committed:
feat: fix docker permission issues

88250 commented on Dockerfile:
```diff
@@ -35,13 +36,34 @@ RUN apk add --no-cache gcc musl-dev && \
+ENV HOME=/home/siyuan
 ENV RUN_IN_CONTAINER=true
 EXPOSE 6806
+VOLUME /home/siyuan/workspace
```
I'm not sure if this will affect users who have specified the `--workspace` parameter, such as the usage described in the README:

```
docker run -d -v workspace_dir_host:workspace_dir_container -p 6806:6806 b3log/siyuan --workspace=workspace_dir_container --accessAuthCode=xxx
```

Macavity replied:
> I'm not sure if this will affect users who have specified the `--workspace` parameter, such as the usage described in the README:
> 
> ```
> docker run -d -v workspace_dir_host:workspace_dir_container -p 6806:6806 b3log/siyuan --workspace=workspace_dir_container --accessAuthCode=xxx
> ```

That's a very valid point. I think I can safely remove this line though, I will test if it impacts the result. 

> Another question, does it work just with docker (not compose)?

Yes, customizing `USER_ID` and `GROUP_ID` is possible with just Docker, not only via Docker Compose. You can pass `--build-arg` during the `docker build` process to specify these values. 

```bash
docker build --build-arg USER_ID=1005 --build-arg GROUP_ID=100 -t siyuan_note_image .
docker run -d --name siyuan_note siyuan_note_image
```

redjumper replied:
I believe that the ownership of the workspace should also change if the workspace is specified.

88250 reviewed (commented):
Another question, does it work just with docker (not compose)?

88250 commented:
In addition, please modify the corresponding document example in README:

```
Alternatively, see below for an example Docker Compose file:

....
```

The above content should be adjusted so that users know how to use it. Thank you.

Macavity committed:
remove VOLUME call on workspace

Macavity commented:
While adjusting the Readme I realized the approach has an issue. It works only for developers who have a copy of the project and will build the image themselves, it does not work for someone using the pre-built image 🤔 

I will have to adjust my approach.

Macavity committed:
move responsibility for user and group creation to entrypoint.sh

Macavity commented:
@88250 I've extracted the logic for user and group creation from the Dockerfile to an `entrypoint.sh` which is called then the container is created initially.

I've updated the english readme and tried to do the same for the chinese readme using DeepL. Can you check if the chinese one makes any sense?

88250 commented:
Great! I'll adjust the details of the Chinese README later. 

Thank you very much for your contribution!
